src/vss_tools/exporters/graphql.py

```diff
@@ -1,134 +1,558 @@
+    """Get GraphQL enums for VSS units and quantity kinds."""
+    global mapping_quantity_kinds_df
+
+    spec_quantity_kinds = get_quantity_kinds_and_units()
+    unit_enums: Dict[str, graphene.Enum] = {}
+
+    # Create a graphene enum for each key in the spec_quantity_kinds
+    for quantity_kind, units in spec_quantity_kinds.items():
+        enum_name = get_unit_enum_name(quantity_kind)
+        enum_values = {}
+        unit_mappings = {}
+        for unit in units:
+            unit_name = get_gql_name(unit, GQLElementType.ENUM_VALUE)
+            unit_mappings[unit] = unit_name
+            enum_values[unit_name] = unit_name
+
+        unit_enums[enum_name] = type(enum_name, (graphene.Enum,), enum_values)  # type: ignore
+        mapping_quantity_kinds_df.loc[quantity_kind] = [enum_name, sort_dict_by_key(unit_mappings)]
+
+    return unit_enums
+
+
+def get_quantity_kinds_and_units() -> dict[str, set[str]]:
+    """Get the quantity kinds and their units as specified in VSS."""
+    spec_quantity_kinds: Dict[str, set[str]] = {}
+    for unit_data in dynamic_units.values():
+        quantity_kind_name = unit_data.quantity
+        if unit_data.unit:
+            unit_name = unit_data.unit
+            if quantity_kind_name not in spec_quantity_kinds:
+                spec_quantity_kinds[quantity_kind_name] = set()
+            spec_quantity_kinds[quantity_kind_name].add(unit_name)
+
+    return dict(sorted(spec_quantity_kinds.items()))
+
+
+def get_branches_with_specified_instances():
+    return vss_branches_df[vss_branches_df["instances"].astype(str) != "[]"]
+
+
+def get_instances_enums() -> Dict[str, graphene.Enum]:
+    """Create a GraphQL enum for each branch that has instances specified."""
+    enums: Dict[str, graphene.Enum] = {}
+    branches_with_instances = get_branches_with_specified_instances()
+
+    for fqn, row in branches_with_instances.iterrows():
+        instance_labels = expand_instances(row["instances"])
+        mapping_instance_labels = {}
+
+        enum_name = get_gql_name(f"{fqn}.Instance", GQLElementType.ENUM)  # TODO: Todo pass a shorter name!
+        enum_values = {}
+
+        for label in instance_labels:
+            value = get_gql_name(label, GQLElementType.ENUM_VALUE)
+            enum_values[value] = value
+            mapping_instance_labels[label] = value
+
+        enums[fqn] = type(enum_name, (graphene.Enum,), enum_values)  # type: ignore
+
+        global mapping_branches_df
+        mapping_branches_df.loc[fqn, ["gql_instance_enum", "instance_labels"]] = [enum_name, mapping_instance_labels]  # type: ignore
+
+    return enums
+
+
+def expand_instances(instances: list) -> list[str]:
+    """Expand the instances specified in the VSS to a list of labels."""
+    inst_levels = []
+    for element in instances:
+        if isinstance(element, list):
+            inst_levels.append(element)
+        elif isinstance(element, str):
+            match = re.match(r"(\w+)\[(\d+),(\d+)\]", element)
```

Comment:
this is the same as tree.py:expand_string() I guess

Reply:
Updated to import the tree.py:expand_string() and reuse the function.